src/NuGet.Core/NuGet.Packaging/PackageCreation/Utility/PackageIdValidator.cs

```diff
@@ -19,6 +19,14 @@ public static class PackageIdValidator
             options: RegexOptions.IgnoreCase | RegexOptions.ExplicitCapture | RegexOptions.CultureInvariant,
             matchTimeout: TimeSpan.FromSeconds(10));
 
+        public static void ValidatePackageIdRegex(string packageId)
+        {
+            if (packageId == null || !IdRegex.IsMatch(packageId))
+            {
+                throw new InvalidPackageIdException(string.Format(CultureInfo.CurrentCulture, Strings.Error_Invalid_package_id, packageId ?? ""));
```

Comment:
I like Kartheek's suggestion, as I was wondering how an ID can be null, and if it is, that should be an argument exception.
If you take that suggestion, then this change can be made as well:

```suggestion
                throw new InvalidPackageIdException(string.Format(CultureInfo.CurrentCulture, Strings.Error_Invalid_package_id, packageId));
```